src/App.jsx

```diff
@@ -1,22 +1,42 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
  import TodoList from './TodoList';
  import AddTodoForm from './AddTodoForm';
+ 
+ function App() {
+  const [todoList, setTodoList] = React.useState(([]));
+  const[isLoading, setIsLoading] = React.useState(true); 
+  
+  React.useEffect (() => {
+    const fetchData = new Promise ((resolve, reject) => {
+      setTimeout (() => {
+        resolve({data: {
+          todoList: JSON.parse(localStorage.getItem(
+          'savedTodoList')) 
+         
+        }  
+      });
+    }, 2000);
+  
+   }); 
+   fetchData.then((result) => {
+    setTodoList(result.data.todoList);
```

Comment:
I tried this out locally, and I noticed that when the local storage is empty (e.g. I started without any todos), the value for `result.data.todoList` was null which caused the app to break. Maybe initialize the local storage to an empty array?